### PR TITLE
Fix show password on login page

### LINKED ISSUE #37432: [Issue] Fix show password on login page
This issue is automatically created based on existing pull request: magento/magento2#37370: Fix show password on login page

---------

### Description (*)

Fix the "Show password" functionality on storefront login page.
### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. Click on Sign in link
2. Type anything on password textbox
3. Click on `Show Password` checkbox

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by thecodecook14**
### Description (*)

Fix the "Show password" functionality on storefront login page.
### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. Click on Sign in link
2. Type anything on password textbox
3. Click on `Show Password` checkbox

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#37432: Fix show password on login page

### CODE REVIEW

**Comment by thecodecook14** [comment]
@magento run all tests

**Comment by thecodecook14** [comment]
@magento run Unit Tests

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Hello @thecodecook14  It seems that the build failures are due to code changes, Please fix them.

File: app/code/Magento/Customer/view/frontend/templates/form/login.phtml [hunk 1]
```diff
@@ -68,7 +68,7 @@
                     "components": {
                         "showPassword": {
                             "component": "Magento_Customer/js/show-password",
-                            "passwordSelector": "#pass"
+                            "passwordSelector": "#pass,#pass1"
```
Comment (Den4ik) [inline app/code/Magento/Customer/view/frontend/templates/form/login.phtml]: In fact initialised component should affect only to form from `login.phtml`.
I agree that we have multiple fields with the same id and in this case make sense to use only one field id or bind by field name. Also please use more appropriate names than `pass1`

File: app/code/Magento/Customer/view/frontend/templates/form/login.phtml [hunk 2]
```diff
@@ -37,7 +37,7 @@
                     <div class="control">
                         <input name="login[password]" type="password"
                             <?php if ($block->isAutocompleteDisabled()): ?> autocomplete="off"<?php endif; ?>
-                               class="input-text" id="pass"
+                               class="input-text" id="pass1"
```
Comment (Den4ik) [inline app/code/Magento/Customer/view/frontend/templates/form/login.phtml]: Please update label `for` attribute to correct input id

**Review by Den4ik — CHANGES_REQUESTED** [review]
@thecodecook14 Thanks for your work.
Please review my comments and look at failed Functional tests

File: app/code/Magento/Customer/view/frontend/templates/form/login.phtml (hunk 1 above)
Comment (thecodecook14) [reply app/code/Magento/Customer/view/frontend/templates/form/login.phtml]: Hello @Den4ik,

First of all, thanks to review this PR. 

I am new in Magento and just trying to contribute here. So to fix the issue we can choose any one option below:

- Use field name - but the name of the field is like this `login[password]`. So how can we pass it in `passwordSelector`?
- Use field class - We can also add a unique class and use that class in `passwordSelector`
- Use field id - This solution is already implemented. As per your suggestion we can change the name of the ID with relevant one. 

Can you please guide me on the first 2 solutions?

**Comment by thecodecook14** [comment]
@magento run all tests

**Comment by thecodecook14** [comment]
@magento give me test instance

**Review by Den4ik — APPROVED** [review]
Changes looks good for me.
@thecodecook14 thanks for work on this PR

**Comment by engcom-Lima** [comment]
@magento create issue

**Comment by engcom-Lima** [comment]
@magento run all tests

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed

**Preconditions:**
- Install fresh Magento **`2.4-develop`** and **`PHP 8.1`**

**Manual testing scenario:**
1. Click on Sign in link
2. Type anything on password textbox
3. Click on `Show Password` checkbox

**Before:** :heavy_multiplication_x: Password was not displaying.
Screenshot 2023-04-26 at 12 24 14 PM


**After:** :heavy_check_mark:  Now password is working fine.
Screenshot 2023-04-26 at 12 30 46 PM

**Additionally tested:**

1. Tested with different kinds of symbols.
2. Done multiple click on the check box.
3. Login with password having different kind of symbols.

No additional manual test case is required as part of regression as it is minor change.

Builds are failed hence, moving this PR to Extended Testing.

**Comment by thecodecook14** [comment]
@magento run WebAPI Tests, Unit Tests

**Comment by engcom-Hotel** [comment]
@magento run Integration Tests, Functional Tests CE

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run Functional Tests EE,Functional Tests B2B

**Comment by engcom-Echo** [comment]
@magento run WebAPI Tests

**Comment by engcom-Echo** [comment]
_Integration Tests_ failure are not related to the PR changes.
_Functional Tests EE, Functional Tests B2B_ test failure are different on last two run on same commit.
**Functional Tests EE**
Run1
image

Run2
image

**Functional Tests B2B**
Run1
image

Run2
image

**Comment by danmooney2** [comment]
Please compare with AC-8823

**Comment by engcom-Charlie** [comment]
This issue has been taken care in one of the internal [jira](https://jira.corp.adobe.com/browse/AC-8823), in [commit](https://github.com/magento/magento2/commit/3d11501a4a9199c808b0a7165911f4ff12ae3d97) revision hence closing this PR now.

**Comment by hostep** [comment]
If someone is looking for an easier patch to apply than that big merge commit mentioned before, this one seems to work in our case: https://github.com/magento/magento2/commit/131303eeeb363a24cbf8fc7749113c4d2e52c354

**Pull request closed without merging**